mantidproject/mantid
# ISIS Powder POLARIS validate chopper mode and make its value persistent

Previously, the `mode` parameter of `isis_powder` (which can have 2 values, `PDF` and `Rietveld`) did not behave as described in the [documentation](http://docs.mantidproject.org/nightly/techniques/ISISPowder-Polaris-v1.html#mode). It was not case-insensitive, and its value was not held in state in the instrument settings.

`mode` was made case-insensitive, and its value was made to persist after it had been set in one method call according to the scheme set out [here](http://docs.mantidproject.org/nightly/techniques/ISISPowder-Tutorials.html#how-objects-hold-state-in-isis-powder). In addition, the nasty behaviour that `Rietveld` was used as the default (so `mode="pdf"` meant that `Rietveld` values were used) was replaced with proper validation.

In addition the system test reference file, which was using the wrong cropping values, was updated.

**To test:**

Point your data directories to `\\olympic\Babylon5\Public\JoeRamsay\POLARIS` and make sure this is at the top of the list. Setup your instrument with the script below.

```
import os
from isis_powder import Polaris

data_dirs = config["datasearch.directories"].split(";")
config_file = os.path.join(data_dirs[0], "polaris_shared_config.yaml")

pol = Polaris(config_file=config_file)
```
First,
```
pol.focus(mode="blah")
```
should fail with the error
```
ValueError: Invalid chopper mode "blah"
```
Now run
```
pol.focus(run_number=100017, mode="PDF")
pol.focus(run_number=100018)
pol.focus(run_number=100019, mode="rietveld")
```
This may take a few minutes.

The workspace `100017-ResultTOF-1` (a member of the group `100017-Results-TOF-Grp`) should be cropped between 1500 and 19900 (ie the PDF cropping values). `100018-ResultTOF-1` should be cropped in the same window.

`100019-ResultTOF-1` should be cropped between 700 and 30000 - the Rietveld cropping values.

Fixes #21350 

**Release Notes** 
[Here](https://github.com/mantidproject/mantid/commit/f52dc3ace5809ede4429fb6a2fbd9ec5599632b5)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
